react-components/src/components/NodeCacheProvider/RevisionFdmNodeCache.ts

```diff
@@ -180,11 +182,40 @@ export class RevisionFdmNodeCache {
     return ancestorMappings.edges;
   }
 
+  public async fetchViewsForAllEdges(): Promise<void> {
+    const allEdgesWithoutView = this.getAllEdges().filter((edge) => edge.view === undefined);
+
+    if (allEdgesWithoutView.length === 0) {
+      return;
+    }
+
+    const nodeInspectionResults = await inspectNodes(
+      this._fdmClient,
+      allEdgesWithoutView.map((edge) => edge.edge.startNode)
+    );
+
+    allEdgesWithoutView.forEach((fdmEdgeWithNode, ind) => {
+      const edgeWithView = {
+        ...fdmEdgeWithNode,
+        view: nodeInspectionResults.items[ind].inspectionResults.involvedViewsAndContainers.views[0]
+      };
+
+      this.insertTreeIndexMappings(edgeWithView.node.treeIndex, edgeWithView);
+    });
+  }
+
   public insertTreeIndexMappings(treeIndex: TreeIndex, edge: FdmEdgeWithNode): void {
     const edgeArray = this._treeIndexToFdmEdges.get(treeIndex);
+    const presentEdge = edgeArray?.find((e) => e.node.id === edge.node.id);
```

Comment:
I'd like to move this to the if-condition below. I'm a bit worried that this will have some bad performance implications, but there will probably not be that many edges associated with a single tree index

Reply:
Yeah, I don't think the amount of edges here will be more than ~10. Moved inside if statement anyway.